spacy/strings.pyx

```diff
@@ -315,3 +315,25 @@ cdef class StringStore:
         self._map.set(key, value)
         self.keys.push_back(key)
         return value
+
+    @cython.boundscheck(False)  # Deactivate bounds checking
+    cdef const unsigned char* utf8_ptr(self, const attr_t hash_val):
+        if hash_val == 0:
+            return b""
+        elif hash_val < len(SYMBOLS_BY_INT):
+            return SYMBOLS_BY_INT[hash_val].encode("utf-8")
+        cdef Utf8Str* string = <Utf8Str*>self._map.get(hash_val)
+        if string.s[0] < sizeof(string.s) and string.s[0] != 0:
+            return string.s[1:string.s[0]+1]
```

Comment:
I think this also creates a wild pointer. Cython slices the C character data, creating a `bytes`, then the `return` will return a pointer to the data. However, since there is no Python reference anymore it will be a wild pointer (from my reading of the Cython docs). The returned data needs to be allocated on the C heap and then deallocated at the call site. Or return a Python `bytes` object or a C++ `std::string` with a copy of the data.

Reply:
OK, thanks, I get the problem. I tried returning a `bytes` object instead of a `char*` but that slows down everything quite significantly. Am I right in thinking that I'd be fine to return `string` itself because it's a pointer to a value in the hash map and thus won't be garbage collected? If so maybe I should do that, deal with extracting the bytes from `string` in the calling code, and come up with some other solution(s) for the special cases (which don't need to be fast because they don't happen very often).

Reply:
That should be fine as I don't see any method on `StringStore` that removes strings from the backing store. But then the lifetime of the `Utff8Str` is bound to the store itself, so you'll need to ensure that it stays alive at least as long as the string pointer needs to.

Reply:
Thanks, the bug should be fixed now. Because the `StringStore` instance is necessary to process the `Doc` object, I don't think there's any situation in which the `Doc` object could outlive it.

Reply:
Changing it like this should theoretically have a small positive impact on speed, although I couldn't measure one.